### ISSUE TITLE
Implement payload persistence offloading for Cassandra

### ISSUE THREAD

**Issue by punktilious** [IBM/FHIR#2900]
**Is your feature request related to a problem? Please describe.**
Relational database systems are not best suited to storing large objects. This can lead to increase runtime costs as well as scalability limitations. 

**Describe the solution you'd like**
Following the groundwork done in #1869, implement payload persistence offloading for Cassandra.

The solution must provide a mechanism for cleaning orphaned records which may remain after a failed transaction.

The solution should consider using asynchronous patterns for storing and fetching the payload data. This will be important when processing large bundles to maintain acceptable response times.

**Describe alternatives you've considered**
Accept the limitations and cost of keeping the payload within the RDBMS.

**Acceptance Criteria**

1.  
   GIVEN a server configured with payload persistence for Cassandra enabled
    AND a new resource is ingested
   WHEN the transaction is successful
   THEN the resource can be read back
    AND the payload can be found in Cassandra

2.  
   GIVEN a server configured with payload persistence for Cassandra enabled
    AND a new resource is ingested
   WHEN the transaction is rolled back
   THEN the payload can not found in Cassandra

3.  
   GIVEN a server configured with payload persistence for Cassandra enabled
    AND a new resource is ingested
   WHEN the transaction fails and rollback is not possible
   THEN the payload is removed from Cassandra after the cleanup process is run

**Additional context**
Design discussion to be had regarding how, where and when to run the transaction cleanup process.

The payload persistence mechanism is not tied to a Liberty Datasource so Liberty's transaction recovery mechanism doesn't apply. The discussion is really about whether the cleanup process should be a separate standalone job, or a background thread 
started by the FHIRServletContext.

Depends on #2899.

**Comment by lmsurpre** [comment]
The initial implementation is done, but we're shifting our focus to the S3/ minio flavor of payload offloading.  We'll need a separate task for documenting how to package the cassandra offloading withe the server and to thoroughly test the Cassandra offloading as part of that effort.
